pkg/proxy/register_test.go

```diff
@@ -117,3 +118,70 @@ func createRegister(r router.Router) *Register {
 
 	return register
 }
+
+func TestParametersReplacement(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	def := &Definition{
```

Comment:
Couldn't this be inside of createProxyDefinitions func?

Reply:
I wanted to have it isolated to reduce as much as possible any external factor influencing the test, do you want me to move it in?

Reply:
Also I would suggest such structure for this tests if possible:

```go
func TestSuccessfulProxy(t *testing.T) {
    t.Parallel()

    log.SetOutput(ioutil.Discard)

    tests := []struct {
        scenario, url, expectedContentType string
        expectedCode                       int
        function                           func(*testing.T, *test.Server, string, string, string, int)
    }{
        {
            scenario:            "Get example route",
            url:                 "/example",
            expectedContentType: "application/json; charset=utf-8",
            expectedCode:        http.StatusOK,
            function:            testGetExampleRoute,
        },
        {
            scenario:            "Get invalid route",
            url:                 "/invalid-route",
            expectedContentType: "text/plain; charset=utf-8",
            expectedCode:        http.StatusNotFound,
            function:            testGetInvalidRoute,
        },
    }

    ts := test.NewServer(createRegisterAndRouter())
    defer ts.Close()

    for _, test := range tests {
        t.Run(test.scenario, func(t *testing.T) {
            test.function(t, ts, test.url, test.expectedContentType, test.scenario, test.expectedCode)
        })
    }
}

func testGetExampleRoute(t *testing.T, ts *test.Server, url, contentType, description string, code int) {
    res, err := ts.Do(http.MethodGet, url, make(map[string]string))
    assert.NoError(t, err)
    if res != nil {
        defer res.Body.Close()
    }

    assert.Equal(t, contentType, res.Header.Get("Content-Type"), description)
    assert.Equal(t, code, res.StatusCode, description)
}

... more tests here
```

Reply:
If they do the same thing better to move them, when reading the code one can assume there's something different with these tests